core/sup_tree_core/termination_manager.ex

```diff
@@ -41,7 +50,12 @@ defmodule AntikytheraCore.TerminationManager do
         cleanup(brokers)
         %State{state | in_service?: false, not_in_service_count: new_count}
       else
-        %State{state | not_in_service_count: new_count}
+        if !log_flushed? && new_count >= @flush_log_threshold_count do
+          flush_gear_logs()
+          %State{state | not_in_service_count: new_count, log_flushed?: true}
+        else
+          %State{state | not_in_service_count: new_count}
+        end
       end
```

Comment:
(optional)
How about avoiding nested `if`?
In addition, I think it might be good to write a log message before starting flushing gear logs if possible.

```elixir
      cond do
        in_service? and new_count >= @threshold_count ->
          L.info("confirmed that this host is to be terminated; start cleanup...")
          VersionUpgradeTaskQueue.disable()
          cleanup(brokers)
          %State{state | in_service?: false, not_in_service_count: new_count}
        !log_flushed? && new_count >= @flush_log_threshold_count ->
          L.info("start flushing gear logs...")
          flush_gear_logs()
          %State{state | not_in_service_count: new_count, log_flushed?: true}
        true ->
          %State{state | not_in_service_count: new_count}
      end
```

Reply:
fixed dac67174bbe78b1a67259a9843b0275bbd6e9723